### ISSUE TITLE
[Bug]: Extraneous scrollbar when selecting text field

### ISSUE THREAD

**Issue by ehrenMSFT** [microsoft/fluentui#31031]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (24) x64 12th Gen Intel(R) Core(TM) i9-12900K
    Memory: 29.93 GB / 63.71 GB
  Browsers:
    Edge: Chromium (123.0.2420.81)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codepen.io/EhrenMSFT/pen/gOyKxeO

### Bug Description

- Run the sample app
- Set your display zoom to 150%
- Click in the text field

## Actual Behavior
Text field is focused, but an unexpected horizontal scrollbar also appears down below.

## Expected Behavior
Text field is focused.

### Requested priority

Normal

### Products/sites affected

Microsoft Power Query

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by sopranopillow** [comment]
I am not able to repro this, could you provide the size of the screen in pixels?

**Comment by ehrenMSFT** [comment]
Sure. It's 1920 x 1080.

![image](https://github.com/microsoft/fluentui/assets/29806264/74ad73c5-8b87-4835-8743-d09b5213bc92)

**Comment by sopranopillow** [comment]
I'm still not able to repro it. I suspected it could be some invisible border or something that appears on focus, but setting the width exactly to before overflowing content and focusing on the textfield doesn't trigger the scrollbar. Could you recreate it maybe setting a div around it that allows scrolling?

**Comment by Narvalex** [comment]
I used the Edge Browser zoom to 150% and it appears indeed at the bottom. It disapears when you change zoom to something less, like 125%.

**Comment by ehrenMSFT** [comment]
> I'm still not able to repro it. I suspected it could be some invisible border or something that appears on focus, but setting the width exactly to before overflowing content and focusing on the textfield doesn't trigger the scrollbar. Could you recreate it maybe setting a div around it that allows scrolling?

If you're having trouble reproing, maybe try a higher zoom level (such as 200%).

Does your monitor have a native resolution of 1920 x 1080? The issue seems related to fractional pixels, so that might be a required part of the repro.

Regarding setting a div around it, I'm not sure exactly what you're asking for.

**Comment by sopranopillow** [comment]
Okay I see it reproing. It looks like my guess was correct, but this happens due to this overflow style:
![image](https://github.com/microsoft/fluentui/assets/5953191/d881ae65-70ab-43c4-95ec-c5012c2dd584)

When the textfield is unfocused we have a width of 398.67, but when we focus we add an outline that's 2px wide causing this:
![image](https://github.com/microsoft/fluentui/assets/5953191/b89ed784-2fa1-424c-9057-5b59d309b344)
and therefore, overflowing the root where the style is applied.

**Comment by ehrenMSFT** [comment]
Glad you were able to repro. Just to clarify: do you agree this a bug when hosting the control inside a fixed-size container? Or do you consider this by design?

**Comment by sopranopillow** [comment]
Hi sorry for the wait, I actually consider this by design. The root is meant to take the size of the input, so no overflow logic should be added. We do this in v9 as well. If the goal is to let the input resize based on scroll height, that should be dealt through JS like (react-textarea-autosize)[https://github.com/Andarist/react-textarea-autosize) does.

**Comment by ehrenMSFT** [comment]
@sopranopillow Thanks for the clarification. While investigating how to resolve this in our codebase, I realized that my repro was not consistent with what we're actually doing. We aren't setting overflow:auto on the root of the TextField. Instead, we're doing that on a div that contains the TextField.

I've updated my codepen with a new, more accurate repro that doesn't depend on any custom styling of the TextField.

If a TextField is contained in a fixed-size, scrollable container, it doesn't seem like the TextField should change its size just because it’s focused.

Can you take a look?

**Comment by ehrenMSFT** [comment]
Hey @sopranopillow. Just following up. Do you consider the new repro I posted a bug, or by design? Thanks!

**Comment by sopranopillow** [comment]
hi @ehrenMSFT, I'm not able to repro this one. I'm following the same steps as before and just to confirm I'm doing it right, I'm doing this:
* zoom to 150%
* make sure the window width is exactly the same as the window input (tried also making it little less and little more)
* focus on the input

maybe a video of the repro would help.

**Comment by ehrenMSFT** [comment]
@sopranopillow Here's a video repro:
https://github.com/microsoft/fluentui/assets/29806264/2ebbb8f7-7c31-476b-8778-936714389877

My steps were:
- Set the Windows "Scale and layout" zoom level to 150%.
- Leave my browser window width maximized.
- Focus the text field.

**Comment by sopranopillow** [comment]
Hi @ehrenMSFT, I've tried to repro this and got some of the other contributors to try it as well and we weren't able to repro it. FWIW I do think this would be a by design case.

**Comment by ehrenMSFT** [comment]
@sopranopillow Ok, thanks for checking. Regarding this being by design, what is the recommendation for hosting these controls inside a fixed-size, scrollable container, since their size changes when they're focused?

We're getting bugs filed against our application, due to customers using various zoom levels that cause these scrollbars to appear.

**Comment by smhigley** [comment]
I was able to repro the issue! For me, it happened on Edge on Windows at 175% zoom. The cause was Edge calculating the `::after` pseudo-element's width as 0.29px larger than the root node 🤦🏻‍♀️.

I'm pushing a fix that adds `overflow: hidden` to the element with the `::after` focus style.

**Comment by ehrenMSFT** [comment]
@smhigley Awesome! Thanks for fixing it.

**Comment by ehrenMSFT** [comment]
I don't think this is considered By Design anymore. There's an open PR out that's intended to fix it.

**Comment by ehrenMSFT** [comment]
Hi @smhigley. Just curious what the status of the fix is. Thanks!

**Comment by robpaveza** [comment]
Hey @ehrenMSFT -- I'm on the Edge team. There are a few things I've noticed here:

 * In your repro case, you have the containing `div` set to `overflow: auto`. This appears to be what is presenting the horizontal scrollbar, not the fluent control; if you set this to `overflow-x: hidden`, then the repro stops.
 * Further, this occurs at zoom levels 125, 150, and 200, but not 175 or 250
 * If I were to guess, there is probably something related to CSS pixel size rounding. If I force the text field control to active (root `div` adds the CSS class `is-active` and the field group `div` changed from `fieldGroup-112` to `fieldGroup-122`), I can see that the fieldGroup div has borders set to 0.533 pixels wide blue an internal width of 398.933. (This element is set to `box-sizing: border-box`). My supposition is that somehow there is a cumulative floating-point error in multiplying these values; these three values are being truncated for display in DevTools where the sum would be 399.999px, but internally the values are summing to > 400px, causing the overflow to trigger.

My colleague Alison has filed a bug against Chromium here to investigate: https://issues.chromium.org/issues/344921696

**Comment by ehrenMSFT** [comment]
> Hey [@ehrenMSFT](https://github.com/ehrenMSFT) -- I'm on the Edge team. There are a few things I've noticed here:

Hi @robpaveza. If I understand correctly, it looks like the Chromium issue was resolved as By Design.

Any further thoughts?
